test.ts

```diff
@@ -258,6 +258,98 @@ describe('Component', () => {
     })
   })
 
+  describe('#setState', () => {
```

Comment:
Can you add test cases for triggering setState in constructor, render(), shouldComponentUpdate(), componentWillUnmount()